File: CPP Language/CPP Language/Aggregation.cpp
// learn c++
// classes aggregation
#include<iostream>
using namespace std;
class Address
{
    public:
    int pincode;
    string city;
    void setA(int pincode,string city)
    {
        this->city = city;
        this->pincode = pincode;
    }
};
class Student
{
    public:
    string name;
    int id;
    Address ad;
    void setS(int id,string name,Address ad)
    {
        this->id = id;
        this->name = name;
        this->ad = ad;
    }
    void show()
    {
        cout<<"Name : "<<name<<"\t id :"<< id<< endl<< "City : "<< ad.city<<" "<<ad.pincode<< endl;
    }
};
int main()
{
    Address ad;
    ad.setA(400043,"Mumbai");
    Student st;
    st.setS(03,"Abhay",ad);
    st.show();

    return 0;
}

File: Java Language/Java Language/Aggregation.java
class javaprogramming
{   
    public static void main(String[] args) 
    {
        System.out.println("main methode invoked");
        Address cmp = new Address("Mumbai","MH","IND",400043);
        Student me = new Student("abhay",03,10,cmp);

        me.display();
    }
}
class Address 
{
    String city,state,country;
    int pincode;

    public Address(String acity,String astate,String acountry,int apincode)
    {
        city = acity;
        state = astate;
        country = acountry;
        pincode = apincode;
    }

}
class Student
{
    String name;
    int id;
    float pointer;
    Address address;

    public Student(String aname,int aid,float apointer,Address add)
    {
        name = aname;
        id = aid;
        pointer = apointer;
        address = add;
    }

    public void display()
    {
        System.out.println("Student Name : "+name+"id no.:"+id);
    }
}